addon-test-support/@ember/test-helpers/-internal/debug-info.ts

```diff
@@ -0,0 +1,169 @@
+import {
+  run,
+  DebugInfo as BackburnerDebugInfo,
+  QueueItem,
+  DeferredActionQueues,
+} from '@ember/runloop';
+import { assign } from '@ember/polyfills';
+
+const PENDING_AJAX_REQUESTS = 'Pending AJAX requests';
+const PENDING_TEST_WAITERS = 'Pending test waiters';
+const SCHEDULED_ASYNC = 'Scheduled async';
+const SCHEDULED_AUTORUN = 'Scheduled autorun';
+
+type MaybeDebugInfo = BackburnerDebugInfo | null;
+
+interface ISettledState {
+  hasPendingTimers: boolean;
+  hasRunLoop: boolean;
+  hasPendingWaiters: boolean;
+  hasPendingRequests: boolean;
+}
+
+interface ISummaryInfo {
+  hasPendingRequests: boolean;
+  hasPendingWaiters: boolean;
+  autorunStackTrace: string | undefined | null;
+  pendingTimersCount: number;
+  hasPendingTimers: boolean;
+  pendingTimersStackTraces: (string | undefined)[];
+  pendingScheduledQueueItemCount: Number;
+  pendingScheduledQueueItemStackTraces: (string | undefined)[];
+  hasRunLoop: boolean;
+}
+
+/**
+ * Retrieves debug information from backburner's current deferred actions queue (runloop instance).
+ * If the `getDebugInfo` method isn't available, it returns `null`.
+ *
+ * @public
+ * @returns {MaybeDebugInfo | null} Backburner debugInfo or, if the getDebugInfo method is not present, null
+ */
+export function getDebugInfo(): MaybeDebugInfo {
+  let debugEnabled = run.backburner.DEBUG === true;
+  let getDebugInfoAvailable = typeof run.backburner.getDebugInfo === 'function';
```

Comment:
Can we extract this into a separate helper function? Specifically in the tests we need to know if the runtimes backburner is capable of providing getDebugInfo independent of if the DEBUG flag is set. 

Maybe `backburnerSupportsDebugInfo()`?

Reply:
I can, sure.